src/utils.ts

```diff
@@ -535,3 +535,8 @@ export const wrapEvent = <T extends Event>(name: EVENT, nativeEvent: T) => {
     cancelable: true,
   });
 };
+
+export const isPrimitive = (val: any) => {
+  const type = typeof val;
+  return val == null || (type !== "object" && type !== "function");
+};
```

Comment:
ignore this. I've moved the helper from a private file to utils because I used it in the PR, but ended up not to. Still, it makes sense to remain here, so left it